PublicMapping/districtbuilder
# Upgrade to React 17

## Overview

Updates to React v17. There should be no functional changes.

### Checklist


### Demo

N/A

### Notes

1. I upgraded Typescript slightly, but I don't think it was necessary; I just bumped it to the latest version that I could without receiving any warnings. 

2. The current version of theme-ui isn't compatible with the new React JSX transform, so we needed to add DISABLE_NEW_JSX_TRANSFORM=true to the client container environment to disable it.

Upgrading to later versions of theme-ui resulted in numerous Typescript errors. We're going to have to upgrade theme-ui to the latest version in order to upgrade to React 18 anyway, and theme-ui 0.15.0 contains breaking changes that I think will incidentally fix the Typescript errors I was seeing (we're on 0.13.1 and the errors started showing up with version 0.14.0), so it didn't seem like the best use of time to look into fixing them at this point.

Anyway, we should include removing `DISABLE_NEW_JSX_TRANSFORM=true` from `docker-compose.yml` to the issue for upgrading to React 18 and include a note about breaking changes in theme-ui v0.15.0.

## Testing Instructions

- Run `./scripts/update` and then give the frontend as thorough of a test as you can -- it should function identically to staging and/or production.

Closes #1177
